src/Nest/CommonAbstractions/ConnectionSettings/ConnectionSettingsBase.cs

```diff
@@ -183,12 +185,13 @@ public TConnectionSettings MapDefaultTypeNames(Action<FluentDictionary<Type, str
 		}
 
         /// <summary>
-        /// Specify which property on a given POCO should be used to infer the id of the document when 
+        /// Specify which property on a given POCO should be used to infer the id of the document when
         /// indexed in Elasticsearch.
         /// </summary>
         /// <typeparam name="TDocument">The type of the document.</typeparam>
         /// <param name="objectPath">The object path.</param>
         /// <returns></returns>
+        [Obsolete("Removed in 6.0.")]
```

Comment:
I'm inclined not to `Obsolete` this as it is likely to be noisy for a lot of users, since it is a commonly used method. Perhaps an XML comment instead?

Reply:
Moved to ///remarks